src/types.ts

```diff
@@ -598,6 +599,7 @@ export class StarknetContract {
         args?: StringMap,
         options: EstimateFeeOptions = {}
     ): Promise<FeeEstimation> {
+        options = copyWithBigint(options); // copy because of potential changes to the object
```

Comment:
If have to add comments that means this code is hard to understand by itself. We shouldn't change the function arguments, but create a new object with the correct block number

Reply:
this is actually old code, just noticed that it was missing the copying part; but I can try to edit it

Reply:
https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/111/commits/4b20ead70affda21c4b2ba9f2ed6296ea33bdc01